### ISSUE TITLE
🚀 Feature: Improve RAM usage of docker image

### ISSUE THREAD

**Issue by ThowZzy** [stonith404/pingvin-share#241]
### 🔖 Feature description

The docker image (0.17.4) uses a lot of ram even thought its not even in use, maybe the database has too big values or something?

![image](https://github.com/stonith404/pingvin-share/assets/61882536/ec6d5cef-f6d0-4539-b226-b630091ba736)

Would be nice if it was using less ram, and when its in use maybe use a bit more.

**Comment by stonith404** [comment]
Thanks for your report. Do you encounter the high usage only since version `0.17.4`? And to be sure, the high memory usage isn't while uploading files?

**Comment by ThowZzy** [comment]
I always noticed this usage a bit higher than other more optimized images. And no I did not upload yet on this install.
To clarify, its not always 400MB, here the current usage, but I think it could still be optimized somehow.
![image](https://github.com/stonith404/pingvin-share/assets/61882536/faf9a99e-a341-437d-b528-407fe1621e6d)

**Comment by stonith404** [comment]
I've checked the memory consumption on my arm64 Linux server and on a Windows 10 PC and I got around 200mb RAM usage which is acceptable in my opinion. 

`docker stats`:
![image](https://github.com/stonith404/pingvin-share/assets/58886915/7e57b19b-09e3-413b-b541-4f3ddcc99594)

`htop inside the container`:
![tempsnip](https://github.com/stonith404/pingvin-share/assets/58886915/1873a2cb-e84f-49df-9479-2626bf1b6b59)


Would you mind to share these two stats too? For the first one you just have to run `docker stats` and for the second one you have to to run the following commands:
```
docker compose pingvin-share bash
apt update && apt install htop
htop
```

**Comment by ThowZzy** [comment]
Almost the same, a bit more

![image](https://github.com/stonith404/pingvin-share/assets/61882536/cd12f4c9-591b-48da-9318-daa50c68fc10)

![image](https://github.com/stonith404/pingvin-share/assets/61882536/3973faf3-3b11-43e3-a492-7a1d16c695f2)

I just find it strange for a simple webpage to be the biggest memory consumption compared to all these other sevices 
![image](https://github.com/stonith404/pingvin-share/assets/61882536/c8643f9c-3fa6-4872-a15e-f99406ed0200)
 But no, its not ridiculously high

**Comment by stonith404** [comment]
Yeah I mean it's still not perfect but in my opinion it's acceptable. Do you think we could convert this bug report into a enhancement?

**Comment by ThowZzy** [comment]
> Yeah I mean it's still not perfect but in my opinion it's acceptable. Do you think we could convert this bug report into a enhancement?

Yes

**Comment by BBergle** [comment]
On version:
Unraid version 6.12.4

When downloading large files my server's RAM completely fills up, is this related?

 
![image](https://github.com/stonith404/pingvin-share/assets/58752312/a0562bb0-d369-40ee-aa8b-3d1937d5b84c)

**Comment by babsz02** [comment]
> Sur la version :0.18.1 version non raidée 6.12.4
> 
> Lors du téléchargement de fichiers volumineux, la RAM de mon serveur se remplit complètement jusqu'à ce que le programme plante, est-ce lié ?
> 
> ![image](https://private-user-images.githubusercontent.com/58752312/270503258-a0562bb0-d369-40ee-aa8b-3d1937d5b84c.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDg5ODQ5ODQsIm5iZiI6MTcwODk4NDY4NCwicGF0aCI6Ii81ODc1MjMxMi8yNzA1MDMyNTgtYTA1NjJiYjAtZDM2OS00MGVlLWFhOGItM2QxOTM3ZDViODRjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAyMjYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMjI2VDIxNTgwNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTZjYjY5YjU5NzFkMGJiZDRlY2FmYmQ1MzRiZDBjZDhjM2NjNTY0MTM3NzM2NGRiODQ3NDVjMDFiOTNkODJkNmEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.6jCXsAkQYFIQ3yU-VzFQXcqpBZs1N8H50PdeHmyTI8c)

I have the same problem 
during upload and download pingvin consumes the entire ram and crashes the program with a permature close

**Comment by babsz02** [comment]
@stonith404 In my case, the premature error occurs when the ram is used to 95% when I download large files.
However, by disabling proxo-buffuring in my swag config, I can download small files without any problem.
The problem comes when uploading and downloading large files.

**Comment by axiomen** [comment]
This might be part of the #354. I'll try to test it again with the 24GB file that caused Premature error on the latest 0.22.2 image.

**Comment by stonith404** [comment]
Thanks, @axiomen. Could you guys try it again, too?
